rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -513,11 +517,31 @@ fn contact_neighbor(peer: &Peer, our_port: u16) -> Result<(), Error> {
 impl TunnelManager {
     pub fn new() -> Self {
         let start = SETTING.get_network().wg_start_port;
-        let ports = (start..65535).collect();
-        TunnelManager {
-            free_ports: ports,
-            tunnels: HashMap::new(),
+        let udp_table = match KI.used_ports() {
+            Ok(used_ports) => used_ports,
+            Err(_) => vec![],
+        };
+
+        let ports: HashMap<u16, bool> = (start..65535)
+            .into_iter()
+            .map(|port| (port, !udp_table.contains(&port)))
```

Comment:
Since `udp_table` could be blank, this could be interpreted as _all ports being open_ which, if `used_ports()` failed then it could certainly not be the case.